### ISSUE TITLE
Do not re-seed NumPy's global RNG whenever possible (NEP-19)

### ISSUE THREAD

**Issue by YuriyGuts** [keras-team/keras#12258]
## Rationale

Global mutable state should be avoided in software because it makes the code less modular and more difficult to reason about. Since one of the guiding principles of Keras is modularity, mutable state should be contained in Keras itself whenever possible, and not affect the global state of larger applications that may import Keras.

According to [NumPy NEP 19 — Random Number Generator Policy](https://www.numpy.org/neps/nep-0019-rng-policy.html), instantiating local `RandomState` objects is preferred, rather than seeding the global `np.random` generator.

> The preferred best practice for getting reproducible pseudorandom numbers is to instantiate a generator object with a seed and pass it around. The implicit global RandomState behind the numpy.random.* convenience functions can cause problems, especially when threads or other forms of concurrency are involved. Global state is always problematic. We categorically recommend avoiding using the convenience functions when reproducibility is involved.

While seeding the global RNG is acceptable in tests, there are some "production" modules in Keras that follow this practice:

1. Dataset loaders (boston_housing, imdb, reuters):

```python
import numpy as np
from keras.datasets import imdb


np.random.seed(123)
print(np.random.randint(0, 100, size=10))
# >>> [66 92 98 17 83 57 86 97 96 47]

np.random.seed(123)
train, test = imdb.load_data()
print(np.random.randint(0, 100, size=10))
# >>> [31 78 85  9 87  2 79 73 52 31]
```

2. Orthogonal initializer.

```python
import numpy as np
from keras.initializers import Orthogonal
from keras.layers import Input, LSTM, Dense
from keras.models import Model


np.random.seed(123)
print(np.random.randint(0, 100, size=10))
# >>> [66 92 98 17 83 57 86 97 96 47]

np.random.seed(123)
initializer = Orthogonal(seed=9876)
layer_input = Input(shape=(50, 5))
layer_lstm = LSTM(units=10, kernel_initializer=initializer, recurrent_initializer=initializer)(layer_input)
layer_output = Dense(1, activation='sigmoid')(layer_lstm)
model = Model(inputs=layer_input, outputs=layer_output)
print(np.random.randint(0, 100, size=10))
# >>> [45 30 65 63 45 71 36 95 69 12]
```

## Expected outcome

If the application code seeds the global RNG, using Keras utilities should not re-seed it (the outputs above should match).

## Tested Environment

- OS: macOS Mojave 10.14.3
- Device: CPU
- Backend: TensorFlow
- Python: 3.7
- Distribution: pip

## Checklist

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by gabrieldemarmiesse** [comment]
That seems reasonable to me. PRs welcome.

**Comment by YuriyGuts** [comment]
@gabrieldemarmiesse Thank you! I've linked a green PR to this issue: https://github.com/keras-team/keras/pull/12259
